From ac5c7df554d5b100ec9a2b137105d7f8297689e4 Mon Sep 17 00:00:00 2001
From: Pascal Terjan <pterjan@google.com>
Date: Mon, 17 Feb 2014 14:44:14 +0000
Subject: [PATCH] Explicitly link to vorbis and ogg

vorbisenc is no longer overlinking and ripper.c uses symbols from both.

https://git.xiph.org/?p=mirrors/vorbis.git;a=commitdiff;h=24b8f4f4c2f2e33333f47738fe5838197ec2a4dc
---
 src/clients/vistest/wscript | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/clients/vistest/wscript b/src/clients/vistest/wscript
index 616d65a..7ec4510 100644
--- a/src/clients/vistest/wscript
+++ b/src/clients/vistest/wscript
@@ -28,7 +28,7 @@ def build(bld):
             target = 'xmms2-ripper',
             source = 'ripper.c',
             includes = '. ../../.. ../../include',
-            uselib = 'vorbisenc',
+            uselib = 'vorbisenc vorbis ogg',
             use = 'xmmsclient'
             )
         if not bld.env.INSTALL_VIS_REFERENCE_CLIENTS:
@@ -53,8 +53,13 @@ def configure(conf):
 
     conf.check_cc(lib="m", uselib_store="math", mandatory=False)
 
-    conf.check_cfg(package='vorbisenc', uselib_store='vorbisenc',
-            args='--cflags --libs', mandatory=False)
+    if conf.check_cfg(package='vorbisenc', uselib_store='vorbisenc',
+            args='--cflags --libs', mandatory=False):
+        conf.check_cfg(package='vorbis', uselib_store='vorbis',
+                args='--cflags --libs', mandatory=True)
+        conf.check_cfg(package='ogg', uselib_store='ogg',
+                args='--cflags --libs', mandatory=True)
+
 
     if conf.check_cfg(package='libvisual-0.4', uselib_store='visual',
             args='--cflags --libs', mandatory=False):
-- 
1.8.5.4

